Scripts/build-sim.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+#
+# Builds the beta Bitwarden iOS app in Debug mode for the iOS Simulator, for ease of automated testing.
+#
+# Usage:
+#
+#   $ ./build-sim.sh
+
+set -euo pipefail
+
+BUILD_DIR="build"
+DERIVED_DATA_PATH="${BUILD_DIR}/DerivedData"
+
+echo "🧱 Building in $(pwd)"
+echo "🧱 Derived Data path ${DERIVED_DATA_PATH}"
+echo ""
+
+echo "🌱 Generating Xcode project"
+mint run xcodegen
+
+mkdir -p "${BUILD_DIR}"
+
+echo "🔨 Performing Xcode build"
+xcrun xcodebuild \
+  -project Bitwarden.xcodeproj \
+  -scheme Bitwarden \
+  -configuration Debug \
+  -destination "platform=iOS Simulator,OS=18.1,name=iPhone 16" \
```

Comment:
It doesn't actually matter, it just needs something specified. If you just say `platform=iOS Simulator` it indicates that it can't find a simulator that matches those requirements. But the resulting binary isn't to the best of my knowledge constrained to that particular OS or device.

Would it possibly make sense to just pull it from the `.test-*` files anyways, so we don't accidentally run into issues with that later on?

Reply:
Great idea 💡 It would be awesome to pull the OS name + version from the ` .test-*` files since that's the "main" destination when developing and it will always be updated "automatically"👏

Reply:
I think that's what I was originally planning on doing and then forgot partway through working this 😅 Should be pretty easy to pull that logic from the `test.yml` workflow